### ISSUE TITLE
Faust's schema mangenment is kinda inefficient.

### ISSUE THREAD

**Issue by rightx2** [faust-streaming/faust#384]
I've skipped all required fields to raise issues because this is about the way faust work,not a bug.

Every time I made a stream agent(or app), I have to define a data class inheriting `faust.Record`.

In my opinion, this increase schema management cost here is why:

Let's say the producer sends data to Kafka and stream data processing is required using data from the Kafka.

In most cases, schema is defined in the producer side and registered in the schema registry before sending data to kafka. Schema registry here is a centralized server across the apps, managing various schema using kafka internal topic.

But in a current way, Faust must define its own brand new data class in the code, inheriting `faust.Record`, which could raise issues like inconsistent schemas between schema registry and the data type class defined in the code . Moreover, every time the schema changed in producer side, I always have to fix Faust code(the data record(class) fields) for compatibility.

Is there any way I can use faust without defining the data class inheriting `faust.Record`?

Related question: https://stackoverflow.com/questions/74062969/is-there-any-way-to-define-app-table-without-using-record-in-faust

**Comment by thomas-chauvet** [comment]
Hello,

I can suggest to keep consistency between schema registry and your `faust Record` models to use the library [
Dataclasses Avro Schema ](https://marcosschroh.github.io/dataclasses-avroschema/faust_records/).

You can get inspiration in this [blog post](https://marcosschroh.github.io/posts/schema-registry-client/) to use avro schema with Faust thanks to the codec registry (you can check [here](https://marcosschroh.github.io/python-schema-registry-client/faust/) to register codec).

In summary, the idea is to use your user-defined faust record models to get also avro schema compatible with the schema registry. Thus, you can directly consume/produce messages in avro with the stack Kafka, Faust and schema-registry.

Hope the resource will help you.

**Comment by rightx2** [comment]
@thomas-chauvet Thanks for answering. Those helped! One more question: Can I use the model calss with pydantic as well?  When I tried 
```
from pydantic import BaseModel

class UserAdvance(faust.Record, AvroModel, BaseModel):
     ....

```
It raised errors when instantiating the class... : (

**Comment by thomas-chauvet** [comment]
I didn't manage to use `faust.Record` and `pydantic.BaseModel` together. I gave up on using pydantic and faust together  in combination with avroschema. Sorry if I cannot help here. You can open an issue on the [repo](https://github.com/marcosschroh/dataclasses-avroschema). The library is quite powerful and not everything is documented maybe the author can help you. He's very reactive.

**Comment by rightx2** [comment]
@thomas-chauvet Thanks thomas, I will raise the issue on the repo you mentioned!

**Comment by wbarnha** [comment]
Thank you @thomas-chauvet for supplying this information, this is a great example and I think it should be added to the documentation.
